### PR TITLE
Contrib/CDS Flashpoint Add Headers

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
1. Add headers in the requests to the API.
2. Resolve the non-highlighting keyword in the layout issue.
3. Update script as per the latest linting standards.
4. Replace the integration image with a one with dimension 120x50.
5. Change human-readable deep link for URL, email, filename and common indicator lookup commands to use the better and more flexible query parameter "query".
6. Introduce 3rd type of IP ("ip-dst|port") in the IP command and its human-readable deep link.
7. Removed "keyword" from domain query for it to be able to match domains flexibly.
8. Update docker image to demisto/python3:3.10.4.29342.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by MLainer1 — CHANGES_REQUESTED** [review]
Good work :), Let's schedule a short demo

File: Packs/Flashpoint/Integrations/Flashpoint/Flashpoint.py [hunk 1]
```diff
@@ -897,12 +921,36 @@ def prepare_incidents_from_compromised_credentials_data(response: dict, next_run
     return next_run, incidents
 
 
+def remove_space_from_args(args):
+    """Remove space from args."""
+    for arg in args:
+        if isinstance(args[arg], str):
+            args[arg] = args[arg].strip()
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/Flashpoint/Integrations/Flashpoint/Flashpoint.py]: Yes, arg is a dictionary. Great suggestion, we should do this.

**Comment by MLainer1** [comment]
We'll schedule a demo after the new changes will be pushed and reviewed

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Hi, 
I added a little suggestion. 

Please let me know when you will be done adding your additional changes.

File: Packs/Flashpoint/Integrations/Flashpoint/Flashpoint.py [hunk 2]
```diff
@@ -1838,17 +1874,18 @@ def get_event_by_id_command(client, event_id):
...
 
     :param client: object of client class
-    :param limit: limit of the records
-    :param report_fpid: report fpid to fetch events
-    :param attack_ids: array of attack ids of event
-    :param time_period: time period i.e 2M, 3d etc
+    :param args: demisto args
     :return: command output
     """
+    limit = args.get('limit', 10)
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/Flashpoint/Integrations/Flashpoint/Flashpoint.py:1885]: Invalid values for the argument limit are handled on the API side. The error message shown is more verbose than the one we get from arg_to_number. Considering that, and the updated standard of having less validation on the XSOAR side, shall we make this change?
image

[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/Flashpoint/Integrations/Flashpoint/Flashpoint.py:1886]: The API will have takes limit as a query parameter which will not differentiate between an integer and a string. Therefore, it is not necessary for us to convert the value into a python integer (we would do it if the limit is passed in the body or perform any validation).

**Review by ShacharKidor — APPROVED** [review]
Hi @crestdatasystems,
Great Job!

Let's have a demo - I can schedule it for tomorrow (Thursday) or for Sunday at 17:00 PM (Israel local time), which option do you prefer?

**Comment by ShacharKidor** [comment]
A Demo was scheduled for Monday (30.05) at 16:00 pm (Israel local time).

File: Packs/Flashpoint/ReleaseNotes/1_3_2.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+##### Flashpoint
+- Added the platform and version headers in the API request.
+- Resolved issue where keywords were not highlighted in the alert incident layout. 
+- Refactored code to meet linting standards.
+- Updated the Docker image to: *demisto/python3:3.10.4.29342*.
+- Changed the human-readable deep link for url, email, filename and common indicator lookup commands to use the flexible "query" parameter.
+- Removed exact match for domain command by removing string "keyword" from the domain query.
+- Introduced the third type of ip (ip-dst|port) in ip command query and human-readable deeplink.
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/Flashpoint/ReleaseNotes/1_3_2.md:10]: Okay, will add that.

**Review by MLainer1 — APPROVED** [review]

**Pull request merged**